packages/clerk-js/src/ui/hooks/useFetch.ts

```diff
@@ -10,35 +68,56 @@ export const useFetch = <T>(
     onSuccess?: (data: T) => void;
   },
 ) => {
-  const [data, setData] = useSafeState<T | null>(null);
-  const requestStatus = useLoadingStatus({
-    status: 'loading',
-  });
+  const cache = useCache(params);
```

Comment:
It appears that `params` is regularly an object here that is not memoized. If I'm understanding the `useCache` implementation correctly, we're using the raw `params` value as a cache key. This means that every re-render could potentially introduce a new entry in the cache. We likely need to stringify the object in a deterministic way (e.g. sorted) to get a consistent cache key (I see we're doing it below).

Additionally, the `useCallback` usage within `useCache` is not so useful for the same reason (`params` changing reference on each call).

Reply:
Thanks for caching this @BRKalow, shall i go the stringify approach ?